File: java_programming_construct/Day5/HarmonicNumber.java
package java_programming_construct.basic_core_java_functional_program;

import java.util.Scanner;

//PROBLEM 4(Harmonic Number)
public class HarmonicNumber {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter N:");
        int N = sc.nextInt();

        if (N == 0) {
            System.out.println("N must not be zero");
            return;
        }

        double sum = 0.0;
        for (int i = 1; i <= N; i++) {
            sum += 1.0 / i;
        }

        System.out.println("Harmonic value = " + sum);
    }
}

File: web_intro/ProgrammingConstruct/ForLoopProblems/HarmonicNumber.js
// Program to find nth Harmonic Number

let n = 5;
let harmonic = 0;

for (let i = 1; i <= n; i++) {
    harmonic = harmonic + (1 / i);
}

console.log("The " + n + "th Harmonic Number is: " + harmonic);